daemon/cmd/daemon.go

```diff
@@ -313,7 +313,7 @@ func createPrefixLengthCounter() *counter.PrefixLengthCounter {
 // then it attempts to clear all IPs from the interface.
 func removeOldRouterState(ipv6 bool, restoredIP net.IP) error {
 	l, err := netlink.LinkByName(defaults.HostDevice)
-	if errors.As(err, &netlink.LinkNotFoundError{}) && restoredIP == nil {
```

Comment:
Per my last comment, wont this only ever be called when restoredIP != nil?

Reply:
With this PR, yes. But this was not the case before this PR. Before this PR, it could happen that `restoredIP` did not match the pod CIDR or native routing CIDR and thus was set to `nil` and we removed all IPs from the `cilium_host` device.

In the new version, we call this _after_ we re-allocated the `restoredIP`, so we know it's valid and not nil. And we call unconditionally  both for restoration and the first time Cilium is started, so the `cilium_host` device might not exist yet.